Client/sdk/game/CModelInfo.h

```diff
@@ -247,6 +248,25 @@ class CModelInfo
     // Vehicle towing functions
     virtual bool IsTowableBy(CModelInfo* towingModel) = 0;
 
+    // 2dfx functions
+    virtual C2DEffectSAInterface* Add2DFXEffect(const CVector& position, const e2dEffectType& type) = 0;
+    virtual bool                  Remove2DFX(C2DEffectSAInterface* effect, bool includeDefault) = 0;
+    virtual bool                  Remove2DFXEffectAtIndex(std::uint32_t index, bool includeDefault = false) = 0;
+    virtual bool                  RemoveAll2DFXEffects(bool includeDefault = false) = 0;
+
+    virtual C2DEffectSA* Get2DFXFromIndex(std::uint32_t index) = 0;
```

Comment:
maybe const?

Reply:
GetFromIndex changes the value of m_pInterface (same as GetInterface function)